### PR TITLE
Added LED Toggle

### PR DESCRIPTION
**Pull request by BRF321JOJO**

### CODE REVIEW

File: src/chisel/lib/uart/Uart.scala [hunk 1]
```diff
@@ -190,13 +194,36 @@ class Echo(frequency: Int, baudRate: Int) extends Module {
   tx.io.channel <> rx.io.channel
 }
 
-class UartMain(frequency: Int, baudRate: Int) extends Module {
+/** Combine RGB and Input/Output. */
+class LEDInput(frequency: Int, baudRate: Int) extends Module {
```
Comment (ekiwi) [inline src/chisel/lib/uart/Uart.scala]: Please copy this over to your application file, i.e. `src/design/LEDToggle.scala `.

Comment (ekiwi) [reply src/chisel/lib/uart/Uart.scala]: And restore the old `UartMain`.

File: src/chisel/lib/uart/Uart.scala [hunk 2]
```diff
@@ -10,13 +10,14 @@
 package chisel.lib.uart
 
 import chisel3._
+import chisel3.experimental.Analog
 import chisel3.util._
+import fpga.ip.ice40._
```
Comment (ekiwi) [inline src/chisel/lib/uart/Uart.scala]: This file should not depend on the specific FPGA. It should be a library that can be used by a variety of different FPGAs.

File: src/design/Echo.scala [hunk 3]
```diff
@@ -0,0 +1,29 @@
+// Copyright 2020 The Regents of the University of California
+// released under BSD 3-Clause License
+// author: Kevin Laeufer <laeufer@cs.berkeley.edu>
```
Comment (ekiwi) [inline src/design/Echo.scala]: feel free to put your name here

File: src/design/LEDToggle.scala [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  when (toggle.io.channel.fire()) {
+    //switch
+    when(toggle.io.channel.bits === 114.U) {
+      REDToggle := !REDToggle
+    }
+  }
+  red := REDToggle
+  green := 0.U
+  blue := 0.U
+}
```
Comment (ekiwi) [inline src/design/LEDToggle.scala:70]: Here would be a good place to put your `LEDInput` module.

**Review by ekiwi — CHANGES_REQUESTED** [review]

**Review by BRF321JOJO — COMMENTED** [review]
Fixed!

**Review by ekiwi — APPROVED** [review]
:ship:

**Pull request merged**